apiserver/user/invitation_redeem.go

```diff
@@ -0,0 +1,133 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	userv1 "github.com/appuio/control-api/apis/user/v1"
+	"github.com/appuio/control-api/apiserver/secretstorage"
+)
+
+var _ rest.Connecter = &invitationRedeemer{}
+var _ rest.StandardStorage = &invitationRedeemer{}
+var _ rest.Scoper = &invitationRedeemer{}
+
+type invitationRedeemer struct {
+	secretstorage.ScopedStandardStorage
+	client client.Client
+
+	usernamePrefix string
+}
+
+func (ir *invitationRedeemer) ConnectMethods() []string {
+	return []string{"REDEEM"}
+}
+
+func (ir *invitationRedeemer) NewConnectOptions() (runtime.Object, bool, string) {
+	// Adds the token from the path to the options under the field "token"
+	return &userv1.RedeemOptions{}, true, "token"
+}
+
+// Connect implements the REDEEM method for invitations.
+// It is used to redeem an invitation by a user.
+// The user is identified by the username in the request context.
+// The token is taken from the path.
+// If the invitation is valid, the invitation is marked as redeemed and the user and a snapshot of the users targets are stored in the status.
```

Comment:
```suggestion
// If the invitation is valid, the invitation is marked as redeemed and the user and a snapshot of the user's targets are stored in the status.
```

nit: "user's targets" is not very clear, given the implementation this refers to the target resources referenced in the invitation which is being redeemed.

Reply:
This is misleading. I'll update the comment.

Reply:
The new version is much better, thanks :+1: